packages/core/database/lib/entity-manager.js

```diff
@@ -199,9 +199,9 @@ const createEntityManager = (db) => {
         throw new Error('Nothing to insert');
       }
 
-      await this.createQueryBuilder(uid).insert(dataToInsert).execute();
+      const createdEntries = await this.createQueryBuilder(uid).insert(dataToInsert).execute();
 
-      const result = { count: data.length };
+      const result = { count: data.length, ids: createdEntries.map(({ id }) => id) };
```

Comment:
My bad, I was positive it was working in MySQL as well.
Any workarounds you can come up with?

Reply:
There are options but nothing native with MySQL.

- We could running inserts one by one
- or we could set an operationId in DB to all the inserted rows so you can fetch the ids afterward. We might have to implement this one day.

Reply:
Just out of curiosity, is that `[2]` the number of affected rows or the last inserted ID?
If the latter, maybe we could simply count backwards `n` times, where `n` is the number of entries created?

Reply:
Ids aren't always sequential, especially in clusters and more specifically in multi-master clusters.

Reply:
Good point!

Since that's the case, I'd say `operationId` is the best choice, given that running queries one by one might hurt performance in very large operations.

What do you think?

Reply:
Yes it is the only `valid` option. And if we want to support relations and such in the create/update many we will have to do it.
I don't think we should do this for now as it's not something we need and it's a lot more work. we should document the change and explain the current MySQL restrictions though

Reply:
PS: it looks like [MariaDB supports `RETURNING`](https://mariadb.com/kb/en/insertreturning/) even though it's a MySQL fork. Is there a way for people to use it?

Reply:
We would need to create a dialect to enable it for maria it would be fairly easy as we already support it with the mysql one :) (not for this PR though :D )